### PR TITLE
Users/nigurr/publish test reports as merged

### PR DESCRIPTION
**Pull request by nigurr**
1. Add Merge Test Reports option for Gradle/Maven/Ant build tasks or by default opt it as merged report in these tasks
2. When using merge to single file, upload it as archive

### CODE REVIEW

File: Tasks/ANT/ant.ps1 [hunk 1]
```diff
@@ -187,7 +187,7 @@ if($publishJUnitResultsFromAntBuild)
 	$runTitleMemberExists = CmdletHasMember "RunTitle"
 	if($runTitleMemberExists)
 	{
-		Publish-TestResults -TestRunner "JUnit" -TestResultsFiles $matchingTestResultsFiles -Context $distributedTaskContext -RunTitle $testRunTitle
+		Publish-TestResults -TestRunner "JUnit" -TestResultsFiles $matchingTestResultsFiles -Context $distributedTaskContext -RunTitle $testRunTitle -MergeResults $true
```
Comment (acesiddhu) [inline Tasks/ANT/ant.ps1]: I would rather treat these as default to true when this is not passed? thoughts

File: Tasks/Gradle/Gradle.ps1 [hunk 2]
```diff
@@ -170,7 +170,7 @@ if($publishJUnitResultsFromAntBuild)
 	$runTitleMemberExists = CmdletHasMember "RunTitle"
 	if($runTitleMemberExists)
```
Comment (AshwiniChalla) [inline Tasks/Gradle/Gradle.ps1]: Can we clean this if check?

File: Tasks/Gradle/gradletask.ts [hunk 3]
```diff
@@ -74,7 +74,7 @@ function publishTestResults(publishJUnitResults, testResultsFiles: string) {
     }
 
     var tp = new tl.TestPublisher("JUnit");
-    tp.publish(matchingTestResultsFiles, false, "", "", "", "");
+    tp.publish(matchingTestResultsFiles, true, "", "", "", true);
```
Comment (AshwiniChalla) [inline Tasks/Gradle/gradletask.ts]: Check if this effects our e2e tests?

File: Tasks/ANT/ant.ps1 (hunk 1 above)
Comment (AshwiniChalla) [inline Tasks/ANT/ant.ps1]: Spec of publish test results task is to not merge by default.

**Comment by AshwiniChalla** [comment]
sign off with comments

File: Tasks/Gradle/Gradle.ps1 (hunk 2 above)
Comment (nigurr) [inline Tasks/Gradle/Gradle.ps1]: out of scope for this PR.

File: Tasks/Gradle/Gradle.ps1 [hunk 4]
```diff
@@ -170,7 +170,7 @@ if($publishJUnitResultsFromAntBuild)
...
 	{
-		Publish-TestResults -TestRunner "JUnit" -TestResultsFiles $matchingTestResultsFiles -Context $distributedTaskContext -RunTitle $testRunTitle
+		Publish-TestResults -TestRunner "JUnit" -TestResultsFiles $matchingTestResultsFiles -Context $distributedTaskContext -RunTitle $testRunTitle -MergeResults $true
```
Comment (AshwiniChalla) [inline Tasks/Gradle/Gradle.ps1]: check if this effects our e2e tests. make sure you run e2e tests before checkin.

File: Tasks/Gradle/Gradle.ps1 (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/Gradle/Gradle.ps1]: no harm in cleaning.

File: Tasks/Gradle/Gradle.ps1 (hunk 4 above)
Comment (nigurr) [inline Tasks/Gradle/Gradle.ps1]: Verified. E2E is passing

File: Tasks/ANT/ant.ps1 (hunk 1 above)
Comment (madhurig) [inline Tasks/ANT/ant.ps1]: I would recommend merging by default, when the task was originally written merge was not an option on xplat, so we default to merge false but for performance users would want to merge in most cases.

**Pull request merged**